### PR TITLE
CAttributeFlattener restore old behaviour behind config

### PR DESCRIPTION
**Pull request by MattijsK**
According to the specification, when excluding a parent object and also specializing this. The exclusion should happen AFTER the specialization. Otherwise the specialization will be ignored when the archetype is flattened. This last thing never happened and the specialized objects still appeared in the flattened archetype. This was however "fixed" in #477. 
Although this fix is correct, it introduced backwards incompatible behaviour. 

This PR introduces the following: 
* It restores the old behaviour behind `allowSpecializationAfterExclusion` in the `FlattenerConfiguration`. For now the default will be to restore the old behaviour. But the intention is to flip the default in a major version upgrade of Archie, as it is backwards incompatible.
* It adds a validation warning if you specialize an object after excluding it.

### CODE REVIEW

File: tools/src/test/resources/com/nedap/archie/flattener/siblingorder/openEHR-EHR-CLUSTER.siblingorderchild.v1.0.0.adls [hunk 1]
```diff
@@ -22,9 +22,8 @@ description
 definition
     CLUSTER[id1.1] matches {    -- SiblingOrderChild
         /items matches {
-            -- CLUSTER [id5] (from parent)
-            CLUSTER[id6] occurrences matches {1}     -- Cluster 2
             CLUSTER[id7] occurrences matches {0}     -- Cluster 3
+            CLUSTER[id6] occurrences matches {1}     -- Cluster 2
```
Comment (pieterbos) [inline tools/src/test/resources/com/nedap/archie/flattener/siblingorder/openEHR-EHR-CLUSTER.siblingorderchild.v1.0.0.adls:26]: why this odd reordering?

**Review by pieterbos — COMMENTED** [review]
Looks good so far, see comments for some small things.

File: tools/src/main/java/com/nedap/archie/flattener/FlattenerConfiguration.java [hunk 2]
```diff
@@ -61,6 +61,16 @@ public class FlattenerConfiguration {
      */
     private boolean failOnMissingUsedArchetype = true;
 
+    /**
+     * Only for Operational templates: officially, objects can only be excluded (occurrences matches {0}) after any
```
[the remark this answers is not in the document]
Comment (MattijsK) [reply tools/src/main/java/com/nedap/archie/flattener/FlattenerConfiguration.java]: I think I moved something around for testing. It's back to normal ordering again.

File: tools/src/test/resources/com/nedap/archie/flattener/siblingorder/openEHR-EHR-CLUSTER.siblingorderchild.v1.0.0.adls (hunk 1 above)
Comment (MattijsK) [reply tools/src/test/resources/com/nedap/archie/flattener/siblingorder/openEHR-EHR-CLUSTER.siblingorderchild.v1.0.0.adls:25]: I think I moved something around for testing. It's back to normal ordering again.

**Review by pieterbos — APPROVED** [review]

**Pull request merged**